File: src/main/java/metrotracker/model/BusPosition.java
package metrotracker.model;

import com.google.gson.annotations.Expose;

public class BusPosition {

    @Expose
    private String dateTime;

    @Expose
    private int Deviation;

    @Expose
    private String directionNum;

    @Expose
    private String directionText;

    @Expose
    private double lat;

    @Expose
    private double lon;

    @Expose
    private String tripEndTime;

    @Expose
    private String tripHeadsign;

    @Expose
    private String tripStartTime;

    @Expose
    private String vehicleID;

    public BusPosition() {}
}

File: src/main/ngapp/metrotracker/src/app/shared/BusPosition.ts
export interface BusPosition { //TODO: change Deviation to number when returning from service?
  dateTime: Date;
  Deviation: number;
  directionNum: string;
  directionText: string;
  lat: number;
  lon: number;
  tripEndTime: string;
  tripHeadsign: string;
  tripStartTime: string;
  vehicleID: string;
}
